src/sknnr_spatial/utils/estimator.py

```diff
@@ -0,0 +1,62 @@
+from functools import wraps
+from typing import Callable, Generic
+
+from sklearn.base import BaseEstimator
+from sklearn.utils.validation import NotFittedError, check_is_fitted
+
+from ..image._base import ImageType
+from ..types import AnyType
+from .image import is_image_type
+
+
+class AttrWrapper(Generic[AnyType]):
+    """A transparent object wrapper that accesses a wrapped object's attributes."""
+
+    _wrapped: AnyType
+
+    def __init__(self, wrapped: AnyType):
+        self._wrapped = wrapped
+
+    def __getattr__(self, name: str):
+        return getattr(self._wrapped, name)
+
+    @property
+    def __dict__(self):
+        return self._wrapped.__dict__
```

Comment:
I hadn't seen `wrapt`, but that looks very cool! Since our usage is pretty simple right now I would lean towards keeping our own implementation rather than adding a dependency (especially since it looks like `wrapt` has some C code that might complicate builds). But if we end up needing more advanced functionality, that's definitely worth a closer look.

> is it general purpose enough to be outside the estimator utlities module?

Probably! I just put it there because that's the only place we currently use it, but you're right that there's nothing about it that makes it specific to estimators. I'm happy to move it if you think there's a better place. The only things I'm thinking of are `utils/__init__.py` or make a `utils/_base.py`.

Reply:
I think it's fine to leave it at the same location until we need to use it again or you'd prefer moving it.  I sometimes have a bad habit of burying useful functions in modules where I never look for them again, so probably a reaction to that!  This one is totally your call.

Reply:
Moving if we find another use sounds like a good plan!